File: app/src/main/java/com/harry/rxjava/retrofit/MovieResponse.java
/*
 * Copyright (C) 20015 MaiNaEr All rights reserved
 */
package com.harry.test.rxjava.retrofit;

import java.io.Serializable;
import java.util.List;

/**
 * 类/接口描述
 *
 * @author Harry
 * @date 2016/9/19.
 */
public class MovieResponse implements Serializable {
     String id;
    String alt;
     String year;
     String title;
     String original_title;
     List<String> genres;
     List<Cast> casts;
     List<Cast> directors;
     Avatars images;

    private class Cast implements Serializable{
         String id;
         String name;
         String alt;
         Avatars avatars;

    }

    private class Avatars implements Serializable{
         String small;
         String medium;
         String large;

    }
}

File: rxretrofit/src/main/java/com/harry/rv/rxretrofit/model/MovieResponse.kt
/*
 * Copyright (C) 20015 MaiNaEr All rights reserved
 */
package com.harry.rv.rxretrofit.model

import org.jetbrains.annotations.NotNull
import java.io.Serializable

/**
 * 类/接口描述

 * @author Harry
 */
 class MovieResponse : Serializable {
    var id: String? = null
    var alt: String? = null
    var year: String? = null
    var title: String? = null
    var original_title: String? = null
    @NotNull
    var genres: List<String>? = null
    var casts: List<Cast>? = null
    var directors: List<Cast>? = null
    @NotNull
    var images: Avatars? = null

    inner class Cast : Serializable {
        var id: String? = null
        var name: String? = null
        var alt: String? = null
        var avatars: Avatars? = null
    }

   inner class Avatars : Serializable {
        var small: String? = null
        var medium: String? = null
        @NotNull
        var large: String? = null
    }

    override fun equals(o: Any?): Boolean {
        return this.id == (o as MovieResponse).id
    }
}
